tests/test_decorators.py

```diff
@@ -23,47 +25,88 @@
-    pass
+class DummyDecorator(py_trees.decorators.Decorator):
+    def __init__(self, child, name=py_trees.common.Name.AUTO_GENERATED):
+        super(DummyDecorator, self).__init__(name=name, child=child)
 
 ##############################################################################
 # Tests
 ##############################################################################
 
+def test_set_name():
+    console.banner("Set Name")
+    child = py_trees.behaviours.Success(name="Woohoo")
+    named_decorator = DummyDecorator(name="Foo", child=child)
+    no_named_decorator = DummyDecorator(child=child)
+    print("\n--------- Assertions ---------\n")
+    print("named_decorator.name == Foo")
+    assert(named_decorator.name == "Foo")
+    print("no_named_decorator.name == DummyDecorator\\n[Woohoo]")
+    assert(no_named_decorator.name == "DummyDecorator\n[Woohoo]")
+
+def test_invalid_child():
+    console.banner("Invalid Child")
+    print("\n--------- Assertions ---------\n")
+    print("TypeError is raised")
+    assert_raises(TypeError, DummyDecorator.__init__, child=5)
+
+def test_invalid_setup():
+    console.banner("Invalid Setup")
+    parent = py_trees.decorators.Decorator(
+        name="Decorator",
+        child=InvalidSetup(name="Invalid Setup")
+    )
+    print("\n--------- Assertions ---------\n")
+    print("TypeError is raised")
+    with assert_raises(TypeError) as context:
+        parent.setup(timeout=15)
+    print("TypeError has message with substring 'NoneType'")
+    assert("NoneType" in str(context.exception))
 
 def test_failure_is_success_tree():
     console.banner("Failure is Success Tree")
     root = py_trees.composites.Selector(name="Root")
     failure = py_trees.behaviours.Failure(name="Failure")
-    goon = MustGoOnRegardless(name="DontBeAfraidToBeTheGoon")
+    goon = py_trees.decorators.FailureIsSuccess(
+        child=py_trees.behaviours.Failure()
+    )
     root.add_child(failure)
     root.add_child(goon)
     py_trees.display.print_ascii_tree(root)
     visitor = py_trees.visitors.DebugVisitor()
-    py_trees.tests.tick_tree(root, visitor, 1, 1)
-
+    py_trees.tests.tick_tree(root, 1, 1, visitor, print_snapshot=True)
+ 
     print("\n--------- Assertions ---------\n")
     print("root.status == py_trees.common.Status.SUCCESS")
     assert(root.status == py_trees.common.Status.SUCCESS)
     print("failure.status == py_trees.common.Status.FAILURE")
     assert(failure.status == py_trees.common.Status.FAILURE)
     print("goon.status == py_trees.common.Status.SUCCESS")
     assert(goon.status == py_trees.common.Status.SUCCESS)
-
-
+ 
+ 
 def test_success_is_failure_tree():
     console.banner("Success is Failure Tree")
     root = py_trees.composites.Selector("Root")
     failure = py_trees.behaviours.Failure(name="Failure")
-    going_down = py_trees.meta.success_is_failure(py_trees.behaviours.Success)(name="Going Down")
+    going_down = py_trees.decorators.SuccessIsFailure(
```

Comment:
how about renaming it to success_is_failure?

Reply:
Done